src/components/organisms/AssetActions/Pool/index.tsx

```diff
@@ -54,23 +54,40 @@ export default function Pool({ ddo }: { ddo: DDO }): ReactElement {
   const [showAdd, setShowAdd] = useState(false)
   const [showRemove, setShowRemove] = useState(false)
 
+  const [hasAddedLiquidity, setHasAddedLiquidity] = useState(false)
+  const [poolShare, setPoolShare] = useState<string>()
+  const [totalUserLiquidityInOcean, setTotalUserLiquidityInOcean] = useState(0)
+  const [totalLiquidityInOcean, setTotalLiquidityInOcean] = useState(0)
+
+  const [
+    creatorTotalLiquidityInOcean,
+    setCreatorTotalLiquidityInOcean
+  ] = useState(0)
+  const [creatorLiquidity, setCreatorLiquidity] = useState<Balance>()
+  const [creatorPoolTokens, setCreatorPoolTokens] = useState<string>()
+  const [creatorPoolShare, setCreatorPoolShare] = useState<string>()
   // the purpose of the value is just to trigger the effect
   const [refreshPool, setRefreshPool] = useState(false)
-  // TODO: put all these variables behind some useEffect
-  // to prevent unneccessary updating on every render
-  const hasAddedLiquidity =
-    userLiquidity && (userLiquidity.ocean > 0 || userLiquidity.datatoken > 0)
 
-  const poolShare =
-    price?.ocean &&
-    price?.datatoken &&
-    userLiquidity &&
-    ((Number(poolTokens) / Number(totalPoolTokens)) * 100).toFixed(2)
+  useEffect(() => {
+    const hasAddedLiquidity =
+      userLiquidity && (userLiquidity.ocean > 0 || userLiquidity.datatoken > 0)
+    setHasAddedLiquidity(hasAddedLiquidity)
+
+    const poolShare =
+      price?.ocean &&
+      price?.datatoken &&
```

Comment:
I forgot why we check here for those 2 `price` things. But if something in the effect depends on that, we should add it to dependencies I guess. Right now, everything in here only updates when `userLiquidity` changes, is that enough?

Reply:
true, i'll add price